### PR TITLE
Avoid crash with None tasks

### PR DESCRIPTION
**Pull request by ssbarnea**
As ansible allows tasks to be None, we should also avoid crashing when
encountering them.

Includes regression prevention by assuring we can parse files
with empty tasks.

Fixes: #468

### CODE REVIEW

File: examples/example.yml [hunk 1]
```diff
@@ -47,3 +47,6 @@
     - name: always run
       debug: msg="always_run is deprecated"
       always_run: true
+
+    # empty task is currently accepted by ansible as valid code:
+    -
```
Comment (rafaelfolco) [inline examples/example.yml:52]:
```suggestion
    - name: empty
```
This test is also covered, correct me otherwise.

**Review by rafaelfolco — CHANGES_REQUESTED** [review]
empty name with no action would be a good test case, isn't it?

File: examples/example.yml (hunk 1 above)
Comment (ssbarnea) [reply examples/example.yml:52]: Putting a name there would render the test useless because it will produce a dictionary with a key name in it. We need a real `null` there for testing, not even an empty dictionary would test it correctly.

**Review by greg-hellings — APPROVED** [review]
The `if foo in (bar or []):` is weird syntax to my eyes. Took me a minute to get what it's doing, I've just never seen it used like this in Python before. Once I got my eyes wrapped around it, this looks good.

**Comment by ssbarnea** [comment]
> The `if foo in (bar or []):` is weird syntax to my eyes. Took me a minute to get what it's doing, I've just never seen it used like this in Python before. Once I got my eyes wrapped around it, this looks good.

Yes is weird, probably `if bar and foo in bar` would look better? Even so they both serve the same purpose.

I guess you need a better python knowledge to know that "False or []" returns `[]` and not False (but is evaluated as False if needed).

**Comment by greg-hellings** [comment]
> > The `if foo in (bar or []):` is weird syntax to my eyes. Took me a minute to get what it's doing, I've just never seen it used like this in Python before. Once I got my eyes wrapped around it, this looks good.
> 
> Yes is weird, probably `if bar and foo in bar` would look better? Even so they both serve the same purpose.
> 
> I guess you need a better python knowledge to know that "False or []" returns `[]` and not False (but is evaluated as False if needed).

Yes, I've seen it used in assignment and short-hand execution but hadn't seen it used in a conditional before. I agree that this form is the more commonly encountered conditional guard from my experience.

**Review by greg-hellings — APPROVED** [review]
lgtm

**Review by albinvass — APPROVED** [review]
lgtm
